src/raw_node.rs

```diff
@@ -196,6 +192,63 @@ impl Ready {
     }
 }
 
+/// ReadyRecord encapsulates some needed data from the corresponding Ready.
+#[derive(Default, Debug, PartialEq)]
+struct ReadyRecord {
+    number: u64,
+    // (index, term) of the last entry from the entries in Ready
+    last_entry: Option<(u64, u64)>,
+    snapshot: Snapshot,
+    messages: Vec<Message>,
+}
+
+/// PersistLastReadyResult encapsulates the commit index, committed entries and
+/// messages that are ready to be applied or be sent to other peers.
+#[derive(Default, Debug, PartialEq)]
+pub struct PersistLastReadyResult {
+    commit_index: Option<u64>,
+    committed_entries: Vec<Entry>,
+    messages: Vec<Vec<Message>>,
+}
+
+impl PersistLastReadyResult {
+    /// The current commit index.
+    /// It will be None state if there is no update.
+    /// It is not required to save it to stable storage.
+    #[inline]
+    pub fn commit_index(&self) -> Option<u64> {
```

Comment:
I do not think that we shall check whether there are committed entries by `Option`

Reply:
Hmmm, it is not used to check whether there are committed entries, it is used to inform the application the `commit index` is changed. 
Actually, it implicitly means that this peer is a leader if the `commit index` is changed after persisting the entries.